src/Microsoft.DotNet.Build.Tasks/PackageFiles/codeOptimization.targets

```diff
@@ -0,0 +1,81 @@
+                    DestinationDirectory="$(OptimizationDataForAssemblyDir)"
+                    OverwriteDestination="true" />
+
+    <!-- Copy the compiled assembly into a folder for further processing -->
+    <MakeDir Directories="$(IntermediateOutputPath)/OptimizedAssembly" />
+    <Message Text="Copy @(IntermediateAssembly) to $(IntermediateOutputPath)/OptimizedAssembly" />
+    <Copy SourceFiles="@(IntermediateAssembly)" DestinationFolder="$(IntermediateOutputPath)/OptimizedAssembly" />
+
+    <!-- Delete existing optimization data from the file that came from with the training data -->
+    <Exec Command="$(IBCMergeToolPath) -f -delete -mo $(OptimizationDataForAssemblyDir)/$(AssemblyName).dll $(OptimizationDataForAssemblyDir)/$(AssemblyName).ibc" />
+
+    <!-- Create ibc data for the new assembly -->
+    <Exec Command="$(IBCMergeToolPath) -f -o $(IntermediateOutputPath)/OptimizedAssembly/$(AssemblyName).new.ibc -delete -mo $(OptimizedAssemblyFile) -incremental $(OptimizationDataForAssemblyDir)/$(AssemblyName).dll" />
+
+    <!-- Compact the optimization data in the optimized assembly -->
+    <Exec Command="$(IBCMergeToolPath) -minify -mu $(OptimizedAssemblyFile) -partialNGEN" />
+
+    <!-- Verify that the optimization data has been applied -->
+    <Exec Command="$(IBCMergeToolPath) -mi $(OptimizedAssemblyFile)" />
+
+    <!-- We need to make sure that the assembly that gets packaged is the one with the optimization data -->
+    <ItemGroup>
+      <IntermediateAssembly Remove="@(IntermediateAssembly)"/>
+      <IntermediateAssembly Include="$(IntermediateOutputPath)/OptimizedAssembly/$(AssemblyName).dll"/>
+    </ItemGroup>
+  </Target>
+
+  <!-- We need the OptimizationData package in order to be able to optimize the assembly -->
+  <Target Name="RestoreOptimizationDataPackage" BeforeTargets="CoreCompile"
+          Condition="'$(EnableProfileGuidedOptimization)'=='true' and !Exists('$(OptimizationDataDir)\project.json')">
+
+    <!-- Dynamically create a project.json file used to restore the optimization data-->
+    <Message Text="Generating project.json for optimization data"  Importance="low" />
+    <ItemGroup>
+      <_OptimizationDataJsonLine Include="{&quot;dependencies&quot;: {" />
+      <_OptimizationDataJsonLine Include="&quot;$(OptimizationDataPackageName)&quot; : &quot;$(OptimizationDataVersion)&quot; " />
+      <_OptimizationDataJsonLine Include="},&quot;frameworks&quot;: {&quot;dnxcore50&quot;: {},&quot;net46&quot;: {}}}"/>
+    </ItemGroup>
+
+    <MakeDir Directories="$(OptimizationDataDir)" ContinueOnError="true" />
+    <WriteLinesToFile File="$(OptimizationDataDir)\project.json" Lines="@(_OptimizationDataJsonLine)" Overwrite="true" />
+
+
+    <!-- Restore the OptimizationData package -->
+    <Exec Command="$(DnuRestoreCommand) $(OptimizationDataDir)\project.json --source https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json"
```

Comment:
That list should be fixed as it should only contain nuget and dotnet-core for most cases. For your case @AlexGhiondea you should directly depend on buildtools and nuget.